pkg/controller/provider/web/client.go

```diff
@@ -136,25 +118,9 @@ func (r *ProviderClient) Get(resource interface{}, id string) (err error) {
 //   ProviderNotReadyErr
 //   NotFoundErr
 func (r *ProviderClient) List(resource interface{}, param ...Param) (err error) {
-	err = r.find()
```

Comment:
Don't attempt to fetch the provider resource first and examine the X-Provider header instaed.
Smarter and reduces API requests.